crypto/cms/cms_enc.c

```diff
@@ -56,8 +56,11 @@ BIO *cms_EncryptedContent_init_bio(CMS_EncryptedContentInfo *ec,
         cipher = EVP_get_cipherbyobj(calg->algorithm);
     }
     if (cipher != NULL) {
+        (void)ERR_set_mark();
         fetched_ciph = EVP_CIPHER_fetch(cms_ctx->libctx, EVP_CIPHER_name(cipher),
                                         cms_ctx->propq);
+        (void)ERR_pop_to_mark();
```

Comment:
Perhaps the pop should be done only if the final cipher is non-NULL below (and clear the mark for cipher==NULL) case.

Reply:
Ahh yeah thanks - hadn't used this before.. Fixed now.